dev/com.ibm.ws.jaxrs.2.0.client_fat/test-applications/baseclient/src/com/ibm/ws/jaxrs20/client/BasicClientTest/client/ClientTestServlet.java

```diff
@@ -164,4 +167,55 @@ public void testFlowProgram(Map<String, String> param, StringBuilder ret) {
         c.close();
         ret.append(res);
     }
+
+    /*
+     * Test to make sure uri's with query parameters with empty values do not
+     * get the '=' character stripped away, aka "?param=" should not become "?param"
+     * If the '=' character is stripped, then by default web container will ignore
+     * the param entirely
+     */
+    public void testQueryParam(Map<String, String> param, StringBuilder ret) {
+        String serverIP = param.get("serverIP");
+        String serverPort = param.get("serverPort");
+        String base = "http://" + serverIP + ":" + serverPort + "/" + moduleName + "/BasicClientTest/BasicResource/query";
+        String uri = base + "?param=";
+        String uri2 = base + "?param";
+
+        ClientBuilder cb = ClientBuilder.newBuilder();
+        Client c = cb.build();
+        WebTarget webTarget = c.target(uri);
+        WebTarget webTarget2 = c.target(uri2);
+
+        // verify that the trailing '=' character is not lost when we build the uri
+        assertEquals(uri, webTarget.getUri().toString());
+
+        // try again with no '='
+        // verify that '=' is not added to the uri
+        assertEquals(uri2, webTarget2.getUri().toString());
+
+        //verify that webcontainer recognized the query parameter with an empty value
+        if (!verifyWebContainer(webTarget)) {
```

Comment:
Can we try this same test with `webTarget2` - making sure that the ALLOW_QUERY_PARAM_WITH_NO_EQUAL web container setting is enabled?

Reply:
Setting <webContainer allowqueryparamwithnoequal="true"/> on the server.xml, but it is still not processing the parameter. Investigating further with the webcontainer team.